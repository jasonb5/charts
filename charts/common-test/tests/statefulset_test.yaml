suite: test statefulset
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: statefulset

      minReadySeconds: 10
      ordinals:
        start: 1
      persistentVolumeClaimRetentionPolicy:
        whenDeleted: Delete
      podManagementPolicy: OrderedReady
      replicas: 10
      revisionHistoryLimit: 20
      serviceName: backend
      updateStrategy:
        type: RollingUpdate
        rollingUpdate:
          maxUnavailable: 2
          partition: 2
    asserts:
      - isKind:
          of: StatefulSet
        documentIndex: 0
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should pass
    set:
      workload: statefulset
    asserts:
      - isKind:
          of: StatefulSet
        documentIndex: 0
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 2
