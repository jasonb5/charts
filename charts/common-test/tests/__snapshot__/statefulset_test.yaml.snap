should overwrite defaults:
  1: |
    minReadySeconds: 10
    ordinals: map[start:1]
    persistentVolumeClaimRetentionPolicy: map[whenDeleted:Delete]
    podManagementPolicy: OrderedReady
    replicas: 10
    revisionHistoryLimit: 20
    selector:
      matchlabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
    serviceName: backend
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
      spec:
        containers:
          - image: busybox:0.1.0
            name: common-test
    updateStrategy:
      rollingUpdate:
        maxUnavailable: 2
        partition: 2
      type: RollingUpdate
should pass:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchlabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: common-test
        spec:
          containers:
            - image: busybox:0.1.0
              name: common-test
      updateStrategy:
        type: OnDelete
